src/compiler/scala/tools/nsc/typechecker/Duplicators.scala

```diff
@@ -229,7 +229,12 @@ abstract class Duplicators extends Analyzer {
 
         case ddef @ DefDef(_, _, _, _, tpt, rhs) =>
           ddef.tpt modifyType fixType
-          super.typed(ddef.clearType(), mode, pt)
+          val result = super.typed(ddef.clearType(), mode, pt)
+          // TODO this is a hack, we really need a cleaner way to transport symbol attachments to duplicated methods
+          // bodies in specialized subclasses.
+          if (ddef.hasAttachment[DelambdafyTarget.type])
+            result.symbol.updateAttachment(DelambdafyTarget)
+          result
```

Comment:
ok, so we go through `cloneSymbol` which seems to also clone attachments (https://github.com/scala/scala/blob/25165b7ad9c4aafdddb4e871b195a91014a91b72/src/reflect/scala/reflect/internal/Symbols.scala#L1951) - shouldn't cloned sym already have the `DelambdafyTarget` attachment?